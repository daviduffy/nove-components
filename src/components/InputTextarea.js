// External Dependencies
import React from 'react';
import PropTypes from 'prop-types';

const InputTextarea = ({
  id,
  label,
  placeholder,
  classes,
  error,
  value,
  onChange,
  options,
  style,
  type, // purposely strip from rest operator
  required,
  ...rest
}) => (
  <div className={`F__g${classes ? ` ${classes}` : ''}`} style={style}>
    {
      label &&
      <label htmlFor={id} className={`F__l${required ? ' F__l--req' : ''}`}>
        <span>{label}</span>
      </label>
    }
    <textarea
      className={`F__i F__i--t${error ? ' F__i--err' : ''}`}
      id={id}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      {...rest}
    />
  </div>
);
InputTextarea.propTypes = {
  classes: PropTypes.string,
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  error: PropTypes.bool.isRequired,
  placeholder: PropTypes.string,
  options: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  required: PropTypes.bool,
  style: PropTypes.object,
  value: PropTypes.string.isRequired
};
InputTextarea.defaultProps = {
  label: '',
  classes: '',
  options: false,
  placeholder: '',
  required: false,
  style: {}
};

export default InputTextarea;
