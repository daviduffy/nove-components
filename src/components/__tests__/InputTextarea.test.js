// External Dependencies
import React from 'react';
import { shallow } from 'enzyme';

// Internal Dependencies
import InputTextarea from '../InputTextarea';
import { textarea as textareaProps } from '../../fixtures/input';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<InputTextarea {...textareaProps} />);
});

test('should render InputTextarea correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should not render label in InputTextarea', () => {
  const noLabelProps = {
    ...textareaProps,
    label: ''
  };
  wrapper = shallow(<InputTextarea {...noLabelProps} />);
  expect(wrapper.find('label').exists()).toEqual(false);
});

test('should render with error class when error is present', () => {
  expect(wrapper.find('.F__i--err')).toHaveLength(0);
  const errorProps = {
    ...textareaProps,
    error: true
  };
  wrapper = shallow(<InputTextarea {...errorProps} />);
  expect(wrapper.find('.F__i--err')).toHaveLength(1);
});

test('should render with required class when required', () => {
  expect(wrapper.find('.F__l--req')).toHaveLength(0);
  const requiredProps = {
    ...textareaProps,
    required: true
  };
  wrapper = shallow(<InputTextarea {...requiredProps} />);
  expect(wrapper.find('.F__l--req')).toHaveLength(1);
});

test('should not allow type attribute even if type supplied', () => {
  expect(wrapper.find('textarea')).toHaveLength(1);
  expect(wrapper.find('[type="text"]')).toHaveLength(0);
  const numberProps = {
    ...textareaProps,
    type: 'text'
  };
  wrapper = shallow(<InputTextarea {...numberProps} />);
  expect(wrapper.find('[type="text"]')).toHaveLength(0);
});
