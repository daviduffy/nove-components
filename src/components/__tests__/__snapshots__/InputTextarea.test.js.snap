// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputTextarea correctly 1`] = `
<div
  className="F__g F__g--name"
  style={Object {}}
>
  <label
    className="F__l"
    htmlFor="message"
  >
    <span>
      Input Label
    </span>
  </label>
  <textarea
    autoFocus={false}
    className="F__i F__i--t"
    id="message"
    onChange={[Function]}
    placeholder="Spill the details, honey"
    value="Chicken fatback swine salami. Capicola venison chuck, hamburger picanha meatball ham ball tip kevin tri-tip salami prosciutto."
  />
</div>
`;
